pkg/config/sync.go

```diff
@@ -55,6 +61,53 @@ func SetConfigForTesting(config *controller.OperatorConfiguration) {
 	updatePublicConfig()
 }
 
+func ApplyExternalDWOCConfig(workspace *dw.DevWorkspace, client crclient.Client) (err error) {
+
+	if !workspace.Spec.Template.Attributes.Exists(constants.ExternalDevWorkspaceConfiguration) {
+		return nil
+	}
+
+	ExternalDWOCMeta := v1alpha1.ExternalConfig{}
+
+	err = workspace.Spec.Template.Attributes.GetInto(constants.ExternalDevWorkspaceConfiguration, &ExternalDWOCMeta)
+	if err != nil {
+		return fmt.Errorf("failed to read attribute %s in DevWorkspace attributes: %w", constants.ExternalDevWorkspaceConfiguration, err)
+	}
+
+	if ExternalDWOCMeta.Name == "" {
+		return fmt.Errorf("'name' must be set for attribute %s in DevWorkspace attributes", constants.ExternalDevWorkspaceConfiguration)
+	}
+
+	if ExternalDWOCMeta.Namespace == "" {
+		return fmt.Errorf("'namespace' must be set for attribute %s in DevWorkspace attributes", constants.ExternalDevWorkspaceConfiguration)
+	}
+
+	externalDWOC := &controller.DevWorkspaceOperatorConfig{}
+	namespacedName := types.NamespacedName{
+		Name:      ExternalDWOCMeta.Name,
+		Namespace: ExternalDWOCMeta.Namespace,
+	}
+
+	err = client.Get(context.TODO(), namespacedName, externalDWOC)
+	if err != nil {
+		return fmt.Errorf("could not fetch external DWOC with name '%s' in namespace '%s': %w", ExternalDWOCMeta.Name, ExternalDWOCMeta.Namespace, err)
+	}
+
+	// TODO: It might be better to just always merge rather than suffering performance hit in configsAlreadyMerged() (or maintaing configsAlreadyMerged if we don't use reflect.DeepEqual)
+	if !configsAlreadyMerged(externalDWOC.Config, internalConfig) {
+		mergeInteralConfigWithExternal(externalDWOC.Config)
+	}
+
+	return nil
+}
+
+func mergeInteralConfigWithExternal(externalConfig *controller.OperatorConfiguration) {
+	configMutex.Lock()
+	defer configMutex.Unlock()
+	mergeConfig(externalConfig, internalConfig)
+	updatePublicConfig()
+}
```

Comment:
I think this needs more planning -- this will update the global config for all workspaces based on the last reconciled workspace, which is not what we intend. Instead, we need some way to read the config with a context, which is significantly more complicated and may require refactoring how config is used in general :confused:

Reply:
Good catch - I overlooked the fact that the internal config is shared between all workspaces. It seems like a prerequisite for the current PR is to allow each workspace to have its own configuration.